File: src/main/scala/com/vc/easy/L349.scala
package com.vc.easy

object L349 {
  def intersection(nums1: Array[Int], nums2: Array[Int]): Array[Int] = {
    import scala.collection.mutable
    val set = mutable.HashSet.empty[Int] ++ nums2
    val arr = new mutable.ArrayBuffer[Int]()
    nums1.foreach(x => {
      if(set.contains(x)) arr += x
    })
    arr.distinct.toArray
  }
}

File: src/main/java/com/vc/easy/L349.java
package com.vc.easy;

import java.util.*;

class L349 {
    public int[] intersection(int[] nums1, int[] nums2) {
        List<Integer> res = new ArrayList<>();

        HashSet<Integer> set = new HashSet<>();
        for(int i = 0; i < nums1.length; i++) set.add(nums1[i]);

        for(int i = 0; i < nums2.length; i++) {
            if(set.contains(nums2[i])) {
                res.add(nums2[i]);
                set.remove(nums2[i]);
            }
        }

        int[] r = new int[res.size()];
        int index = 0;
        for(Integer element: res) r[index++] = element;
        return r;
    }
}